### PR TITLE
script to generate config.h and config.asm for ffmpeg lib

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

**Comment by adipose** [comment]
@clsid2:

I wrote this to make it easy to regenerate config.h and config.asm in the future.  Should only be required if we add new libraries or change build tools (for example VS2017 generating `__restrict` vs `restrict` from VS2019).  One "default" that showed up was bcrypt, but I had removed it from config.h.  I went ahead and left it in, now, but it did require adding the .lib to the build.

Other minor changes for `CC_IDENT` and `FFMPEG_CONFIGURATION` to be accurate to the build.

File: src/thirdparty/ffmpeg/configure_ffmpeg-min.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+echo "$(pwd)" | grep -q '[[:blank:]]' &&
+  echo "Out of tree builds are impossible with whitespace in source path." && exit 1
+
+if [ "${4}" == "VS2015" ]; then
```
Comment (clsid2) [inline src/thirdparty/ffmpeg/configure_ffmpeg-min.sh]: VS2015 can be removed

**Comment by clsid2** [comment]
I was previously still using VS2019 16.6, now 16.11. I think the `restrict` support was added in 16.8.

**Comment by adipose** [comment]
Oh, interesting. Actually that whole block is probably not needed for this script. Makefile handles it.

**Comment by adipose** [comment]
Ok, that should be fixed.  I went ahead and allowed it to use `restrict` again.  Fixed another bug where I blocked out code for testing.

**Pull request merged**